future-apps/actor/contracts/SignatureValidator.sol

```diff
@@ -0,0 +1,52 @@
+pragma solidity 0.4.24;
+
+// From github.com/DexyProject/protocol
+// This should probably be moved into aOS: https://github.com/aragon/aragonOS/pull/442
+
+
+library SignatureValidator {
+    // Maybe use 0x's IDs
+    enum SignatureMode {
+        EIP712,
+        GETH,
+        TREZOR
+    }
+
+    /// @dev Validates that a hash was signed by a specified signer.
+    /// @param hash Hash which was signed.
+    /// @param signer Address of the signer.
+    /// @param signature ECDSA signature along with the mode (0 = EIP712, 1 = Geth, 2 = Trezor) {mode}{v}{r}{s}.
+    /// @return Returns whether signature is from a specified user.
+    function isValidSignature(bytes32 hash, address signer, bytes signature) internal pure returns (bool) {
+        if (signature.length != 66) {
+            return false;
+        }
+
+        SignatureMode mode = SignatureMode(uint8(signature[0]));
+
+        uint8 v = uint8(signature[1]);
+        bytes32 r;
+        bytes32 s;
+        assembly {
+            r := mload(add(signature, 34))
+            s := mload(add(signature, 66))
+        }
+
+        // Allow signature version to be 0 or 1
+        if (v < 27) {
+            v += 27;
+        }
+
+        if (v != 27 && v != 28) {
+            return false;
+        }
+
+        if (mode == SignatureMode.GETH) {
+            hash = keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n32", hash));
+        } else if (mode == SignatureMode.TREZOR) {
```

Comment:
Note that Trezor implements the same signing prefix now: https://github.com/ethereum/go-ethereum/issues/14794#issuecomment-425633910